packages/headless/src/features/search/search-slice.ts

```diff
@@ -1,23 +1,46 @@
 import {createReducer} from '@reduxjs/toolkit';
-import {executeSearch} from './search-actions';
-import {getSearchInitialState} from './search-state';
+import {executeSearch, fetchMoreResults} from './search-actions';
+import {getSearchInitialState, SearchState} from './search-state';
+
+type SearchAction = typeof executeSearch | typeof fetchMoreResults;
+
+function handleRejectedSearch(
+  state: SearchState,
+  action: ReturnType<SearchAction['rejected']>
+) {
+  state.error = action.payload ? action.payload : null;
+  state.isLoading = false;
+}
+
+function handleFulfilledSearch(
+  state: SearchState,
+  action: ReturnType<SearchAction['fulfilled']>
+) {
+  state.error = null;
+  state.response = action.payload.response;
+  state.queryExecuted = action.payload.queryExecuted;
+  state.duration = action.payload.duration;
+  state.isLoading = false;
+}
+
+function handlePendingSearch(state: SearchState) {
+  state.isLoading = true;
```

Comment:
Currently, nothing prevents `fetchMoreResults` from being called simultaneously with `executeSearch`. Should some mechanism be added for that?

Reply:
I would not worry too much about this right now. 

But, you are right, we will need to think of some kind of cancellation token for queries launched at the same time.